.github/files/gh-autorelease/files/autorelease.sh

```diff
@@ -2,32 +2,39 @@
 
 set -eo pipefail
 
+: "${GH_TOKEN:?Build argument needs to be set and non-empty.}"
 : "${GITHUB_REF:?Build argument needs to be set and non-empty.}"
 : "${GITHUB_SHA:?Build argument needs to be set and non-empty.}"
-: "${API_TOKEN_GITHUB:?Build argument needs to be set and non-empty.}"
-: "${GITHUB_API_URL:?Build argument needs to be set and non-empty.}"
-: "${GITHUB_REPOSITORY:?Build argument needs to be set and non-empty.}"
+
+if [[ ! -f composer.json ]]; then
+	echo '::error::No composer.json. Did it get excluded from the mirror?'
+	exit 1
+fi
 
 ## Determine tag
-if [[ ! "$GITHUB_REF" =~ ^refs/tags/v?[0-9]+(\.[0-9]+)+(-[a-z0-9._-]+)?$ ]]; then
-	echo "::error::Expected GITHUB_REF like \`refs/tags/v1.2.3\` or \`refs/tags/1.2.3\`, got \`$GITHUB_REF\`"
+ROLLING_MODE=
+if [[ "$GITHUB_REF" =~ ^refs/tags/v?[0-9]+(\.[0-9]+)+(-[a-z0-9._-]+)?$ ]]; then
+	TAG="${GITHUB_REF#refs/tags/}"
+elif [[ "$GITHUB_REF" == "refs/heads/trunk" ]]; then
+	ROLLING_MODE="$(jq -r '.extra.autorelease.rolling-release' composer.json)"
+	if [[ -z "$ROLLING_MODE" ]]; then
+		echo "::notice::Skipping trunk release because autorelease rolling mode is not enabled."
+		exit 0
+	fi
+	TAG="rolling-release"
+else
+	echo "::error::Expected GITHUB_REF like \`refs/tags/v1.2.3\` or \`refs/tags/1.2.3\` or \`trunk\` for rolling releases, got \`$GITHUB_REF\`"
 	exit 1
 fi
-TAG="${GITHUB_REF#refs/tags/}"
 
 ## Check for alphas
-if [[ "$TAG" =~ -(alpha|a\.[0-9]*[02468])$ ]]; then
+if [[ "$TAG" =~ -(alpha|a\.[0-9]*[02468])$ && -z "$ROLLING_MODE" ]]; then
 	echo "Not creating a release for alpha version $TAG"
 	exit 0
 fi
```

Comment:
Instead of adding the `&& -z "$ROLLING_MODE"`, I'd just move the whole block up to inside the tag-push case of the `if` (i.e. between lines 17 and 18 above).

Reply:
Done in https://github.com/Automattic/jetpack/pull/38994/commits/8aefeab9f182f6c7bb9466d15fae3292d2b39d38, thanks!